File: src/workflowsuite/kpi/client/serviceregistry/ServiceEndpointsInfo.java
package workflowsuite.kpi.client.serviceregistry;

import java.util.ArrayList;

public final class ServiceEndpointsInfo {
    public static final ServiceEndpointsInfo EMPTY = new ServiceEndpointsInfo();

    public String deploymentUnitName = "";

    public String serviceKind = "";

    public ArrayList<EndpointConfiguration> endpoints = new ArrayList<>(0);

    public ArrayList<TransportSettings> transportSettigs = new ArrayList<>(0);
}

File: src/workflowSuite/kpi/client/serviceRegistry/ServiceEndpointsInfo.kt
package workflowsuite.kpi.client.serviceregistry

data class ServiceEndpointsInfo(var deploymentUnitName: String, var serviceKind: String, var endpoints:List<EndpointConfiguration>, var transportSettigs: List<TransportSettings>) {
    companion object {
        val Empty = ServiceEndpointsInfo("", "", emptyList(), emptyList() )

        fun CreateInstance() : ServiceEndpointsInfo {
            return ServiceEndpointsInfo("", "", emptyList(), emptyList())
        }
    }
}